components/brave_wallet_ui/panel/async/wallet_panel_async_handler.ts

```diff
@@ -170,26 +172,27 @@ handler.on(PanelActions.cancelConnectToSite.getType(), async (store: Store, payl
 })
 
 handler.on(PanelActions.cancelConnectHardwareWallet.getType(), async (store: Store, txInfo: TransactionInfo) => {
-  const hardwareAccount = await findHardwareAccountInfo(txInfo.fromAddress)
-  if (!hardwareAccount || !hardwareAccount.hardware) {
+  const found = await findHardwareAccountInfo(txInfo.fromAddress)
+  if (!found || !found.hardware) {
     return
```

Comment:
just an aside: do we not need to report this failure to the user?

Reply:
I do not think so because user clicked by Cancel button and if transaction has already been processed and removed we just need to show him the main screen, changed condition to show the main screen